lib/puppet/provider/ssh_authorized_key/parsed.rb

```diff
@@ -42,9 +42,27 @@ def file_perm
     0o600
   end
 
-  def user
-    uid = Puppet::FileSystem.stat(target).uid
-    Etc.getpwuid(uid).name
+  def group_writable_perm
+    0o020
+  end
+
+  def group_writable?(path)
+    path.stat.mode & group_writable_perm != 0
+  end
+
+  def trusted_path
+    return false unless Puppet::FileSystem.exist?(target)
```

Comment:
If the `target` doesn't exist yet, then we never write the file as root.

Reply:
> Maybe you meant to check if the target's parent directory exists?

Yep, missed that somehow... Thanks!